File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def search(node, largest):
            if largest <= node.val:
                nonlocal result
                result += 1
            if node.left:
                search(node.left, max(largest, node.val))
            if node.right:
                search(node.right, max(largest, node.val))


        result = 0
        search(root, root.val)
        return result

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.java
class Solution {
    private int result;

    public int goodNodes(TreeNode root) {
        result = 0;
        search(root, root.val);
        return result;
    }

    private void search(TreeNode node, int largest) {
        if (largest <= node.val) {
            result++;
            largest = node.val;
        }
        if (node.left != null) {
            search(node.left, largest);
        }
        if (node.right != null) {
            search(node.right, largest);
        }
    }
}